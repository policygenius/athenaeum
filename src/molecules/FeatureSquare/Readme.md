### FeatureSquare Example:

```jsx
    <FeatureSquare
      header='Header'
      subheader='Subheader'
      icon='veryPopular'
    />
```
