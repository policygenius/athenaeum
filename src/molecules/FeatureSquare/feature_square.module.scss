.feature-square {
  color: colors('neutral-8');
  padding: ru(2);
  text-align: center;
  display: flex;
  flex-direction: column;
  align-content: center;
  align-items: center;
  height: 100%;
  justify-content: space-between;
}

.icon {
  margin-bottom: ru(1);
}

.subheader {
  margin-bottom: ru(1);
}
