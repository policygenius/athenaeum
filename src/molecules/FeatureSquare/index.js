import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './feature_square.module.scss';
import colors from 'atoms/Color/colors.scss';

import Icon from 'atoms/Icon';
import Text from 'atoms/Text';
import Button from 'atoms/Button';

function FeatureSquare(props) {
  const {
    children,
    className,
    backgroundColor,
    header,
    subheader,
    icon,
    buttonText
  } = props;

  const classes = [
    styles['feature-square'],
    className,
    backgroundColor && colors['neutral-8'],
    colors[`background-${backgroundColor}`],
  ];

  return (
    <div className={classnames(...classes)}>
      <Icon icon={icon} className={ styles['icon'] }/>
      <Text type={4} semibold>{ header }</Text>
      <Text type={6} className={ styles['subheader'] }>{ subheader }</Text>
      <Button>{ buttonText }</Button>
    </div>
  );
}


FeatureSquare.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,
};

FeatureSquare.defaultProps = {
  // Place any default props here.
  buttonText: 'Get Quotes'
};

export default FeatureSquare;
