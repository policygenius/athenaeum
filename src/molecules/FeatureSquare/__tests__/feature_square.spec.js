import React from 'react';
// import sinon from 'sinon';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FeatureSquare from 'molecules/FeatureSquare';

describe('<FeatureSquare />', () => {
  // let defaultProps;
  //
  // beforeEach(() => {
  //   defaultProps = {
  //     // Add default props here
  //   };
  // });

  it('renders correctly', () => {
    const wrapper = mount(<FeatureSquare />);

    expect(wrapper.type()).toEqual(FeatureSquare);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<FeatureSquare />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
