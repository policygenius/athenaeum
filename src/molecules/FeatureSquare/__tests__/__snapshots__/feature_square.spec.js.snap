// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeatureSquare /> matches snapshot 1`] = `
<div
  className="feature-square background-undefined"
>
  <span
    className="icon-wrapper icon"
    style={
      Object {
        "height": undefined,
        "width": "100px",
      }
    }
  >
    <img
      alt="xIcon"
      className="img"
      src="https://static.policygenius.com/svgs/xIcon.svgz"
      title="xIcon"
    />
  </span>
  <p
    className="primary-3 type-a-5-bold inherit"
  >
    Header!
  </p>
  <p
    className="primary-3 subheader type-b-8-medium inherit"
  >
    Subheader!!!
  </p>
  <div
    className="layout button-wrapper"
  >
    <div
      className="col col-sm-6 border-neutral-6"
    >
      <button
        className="button outline button"
        type="button"
      >
        CLICK SOMETHING ALREADY!
      </button>
    </div>
  </div>
</div>
`;
